extensions/color-shades/src/components/generate-color-shades/generate-color-shades-grid.component.tsx

```diff
@@ -0,0 +1,53 @@
+import { ActionPanel, Action, Icon, Grid } from "@raycast/api";
+import { GeneratorService } from "../../services/generator.service";
+import { PaletteService } from "../../services/palette.service";
+import { usePromise } from "@raycast/utils";
+import { useState } from "react";
+import { Palette } from "../../models/palette.model";
+
+export function GenerateColorShadesGrid(props: { color: string }): JSX.Element {
+  const [baseColor, setBaseColor] = useState<string>(props.color);
+  const { data: palette, isLoading, revalidate } = usePromise(GeneratorService.generatePalette, [baseColor]);
+  return (
+    <Grid isLoading={isLoading} columns={11} navigationTitle={palette?.name}>
+      {Object.entries(palette?.colors ?? []).map(([name, color]) => (
+        <Grid.Item
+          key={name}
+          content={{
+            color: {
+              light: color,
+              dark: color,
+              adjustContrast: false,
+            },
+          }}
+          title={name}
+          subtitle={color}
+          actions={
+            <ActionPanel>
+              <Action.CopyToClipboard content={color} />
+              <Action
+                title="Use as Base Color"
+                icon={{ source: Icon.RotateClockwise }}
+                onAction={() => {
+                  setBaseColor(color);
+                  revalidate();
+                }}
+              />
+              <Action.CopyToClipboard
+                title="Copy All Colors as Variable Declaration Ready"
```

Comment:
Maybe this could be just `Copy All Colors as Variables`? It's currently too long for the menu. You can always add explanations on the `README` if you feel it needs more context.

Reply:
All done, I don't think this needs more explanations it's self-explanatory :)
Thanks a lot for your review!